pkg/config/legacy/converter.go

```diff
@@ -132,44 +137,120 @@ func FromAgentConfig(agentConfig Config) error {
 	//Trace APM based configurations
 
 	if agentConfig["apm_enabled"] != "" {
-		if enabled, err := isAffirmative(agentConfig["apm_enabled"]); err == nil && !enabled {
+		if enabled, err := isAffirmative(agentConfig["apm_enabled"]); err == nil {
 			// apm is enabled by default, convert the config only if it was disabled
 			configConverter.Set("apm_config.enabled", enabled)
 		}
 	}
 
-	if agentConfig["env"] != "" {
-		configConverter.Set("apm_config.env", agentConfig["env"])
+	if agentConfig["trace.config.env"] != "" {
+		configConverter.Set("apm_config.env", agentConfig["trace.config.env"])
 	}
 
-	if receiverPort, err := strconv.Atoi(agentConfig["receiver_port"]); err == nil {
+	if receiverPort, err := strconv.Atoi(agentConfig["trace.receiver.receiver_port"]); err == nil {
 		configConverter.Set("apm_config.receiver_port", receiverPort)
 	}
 
 	if agentConfig["non_local_traffic"] != "" {
-		if enabled, err := isAffirmative(agentConfig["non_local_traffic"]); err == nil && enabled {
+		if enabled, err := isAffirmative(agentConfig["non_local_traffic"]); err == nil {
 			// trace-agent listen locally by default, convert the config only if configured to listen to more
 			configConverter.Set("apm_config.apm_non_local_traffic", enabled)
 		}
 	}
 
-	if sampleRate, err := strconv.ParseFloat(agentConfig["extra_sample_rate"], 64); err == nil {
+	if sampleRate, err := strconv.ParseFloat(agentConfig["trace.sampler.extra_sample_rate"], 64); err == nil {
```

Comment:
why are these trace options handled both here and in `extractTraceAgentConfig`?

Reply:
Accident, there were a couple of duplicates. I've moved everything trace related into `extractTraceAgentConfig` and removed duplicates.